repository/src/test/java/org/alfresco/repo/security/authentication/identityservice/IdentityServiceJITProvisioningHandlerTest.java

```diff
@@ -0,0 +1,166 @@
+    {
+        assertFalse(personService.personExists(IDS_USERNAME));
+
+        IdentityServiceFacade.AccessTokenAuthorization accessTokenAuthorization =
+            identityServiceFacade.authorize(IdentityServiceFacade.AuthorizationGrant.password(IDS_USERNAME, "password"));
+
+        String accessToken = accessTokenAuthorization.getAccessToken().getTokenValue();
+        IdentityServiceFacade idsServiceFacadeMock = mock(IdentityServiceFacade.class);
+        when(idsServiceFacadeMock.decodeToken(accessToken)).thenReturn(null);
+        when(idsServiceFacadeMock.getUserInfo(accessToken)).thenReturn(identityServiceFacade.getUserInfo(accessToken));
+
+        // Replace the original facade with a mocked one to prevent user information from being extracted from the access token.
+        Field declaredField = jitProvisioningHandler.getClass()
+            .getDeclaredField("identityServiceFacade");
+        declaredField.setAccessible(true);
+        declaredField.set(jitProvisioningHandler, idsServiceFacadeMock);
+
+        Optional<OIDCUserInfo> userInfoOptional = jitProvisioningHandler.extractUserInfoAndCreateUserIfNeeded(
+            accessToken);
+
+        declaredField.set(jitProvisioningHandler, identityServiceFacade);
+
+        NodeRef person = personService.getPerson(IDS_USERNAME);
+
+        assertTrue(userInfoOptional.isPresent());
+        assertEquals(IDS_USERNAME, userInfoOptional.get().username());
+        assertEquals("John", userInfoOptional.get().firstName());
+        assertEquals("Doe", userInfoOptional.get().lastName());
+        assertEquals("johndoe@test.com", userInfoOptional.get().email());
+        assertEquals(IDS_USERNAME, nodeService.getProperty(person, ContentModel.PROP_USERNAME));
+        assertEquals("John", nodeService.getProperty(person, ContentModel.PROP_FIRSTNAME));
+        assertEquals("Doe", nodeService.getProperty(person, ContentModel.PROP_LASTNAME));
+        assertEquals("johndoe@test.com", nodeService.getProperty(person, ContentModel.PROP_EMAIL));
+        verify(idsServiceFacadeMock).decodeToken(accessToken);
+        verify(idsServiceFacadeMock).getUserInfo(accessToken);
+    }
+
+    @After
+    public void tearDown()
+    {
+        AuthenticationUtil.runAsSystem(new AuthenticationUtil.RunAsWork<Void>()
+        {
+            @Override
+            public Void doWork() throws Exception
+            {
+                transactionService.getRetryingTransactionHelper()
+                    .doInTransaction(new RetryingTransactionHelper.RetryingTransactionCallback<Void>()
+                    {
+                        @Override
+                        public Void execute() throws Throwable
+                        {
+                            personService.deletePerson(IDS_USERNAME);
+                            return null;
+                        }
+                    });
+                return null;
+            }
+        });
```

Comment:
Even the tests are affected by the aspectj issue? Or can we use lambdas+diamond operators here?

Reply:
Tests are not affected. Changed to the lambda+diamond operator.